### ISSUE TITLE
Support png images for cover art.

### ISSUE THREAD

**Issue by orbea** [Alexey-Yakovenko/deadbeef#1564]
Edit: This is experienced with `deadbeef-8214e4f_2016.04.22_master-x86_64-1_git`.

Is it possible that png files could be supported for cover art as well? Some bandcamp artists provide music with cover art already in the png format and its a chore to convert them to jpg. Other music players like fb2k do not have this issue.

**Comment by Alexey-Yakovenko** [comment]
why do you think that PNG is unsupported?

**Comment by Alexey-Yakovenko** [comment]
Looks like a duplicate of #556 and #345

**Comment by orbea** [comment]
Here is a artist that releases music under a creative commons license.
https://lingouf.bandcamp.com/

The albums that display album art correctly all provide jpg files while the albums that do not provide png files.

Works:
`Internazional fratricide`
`Jours de Glace`
`Kovoclak`
`Kàtàhpyà`
`Lucélma`
`Princess des Douces Hardeucores`
`Return to Condé sur Noireau with Jean-Marie and the chemtrail orchestra`
`Shaqutoussine`

Doesn't work:
`Ecritomate`
`FINS GOURMETS`
`PERLES DE NAUSÉE`

Example of the cover art files.
```
$ ls Internazional\ fratricide/*.jpg
Internazional fratricide/cover.jpg
Internazional fratricide/lingouf - Internazional fratricide - jaquette-recto-cd.jpg
Internazional fratricide/lingouf - Internazional fratricide - macaron.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge1.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge2.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge3.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge4.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge5.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge6.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge7.jpg
Internazional fratricide/lingouf - Internazional fratricide - poge8.jpg

$ ls FINS\ GOURMETS/*.png
FINS GOURMETS/lingouf - FINS GOURMETS - cover.png
FINS GOURMETS/lingouf - FINS GOURMETS - image-fin-gourmet-lacrymale.png
FINS GOURMETS/lingouf - FINS GOURMETS - image-fin-gourmet-nu.png
FINS GOURMETS/lingouf - FINS GOURMETS - image-fin-gourmet-purge.png
FINS GOURMETS/lingouf - FINS GOURMETS - image-fin-gourmet-re-loups-sans-re-morts.png
FINS GOURMETS/lingouf - FINS GOURMETS - image-fin-gourmet-revelatrices.png
```
Here is another bandcamp artist that provide gratis music and has a similar issue.
https://tentacle666.bandcamp.com/
```
$ ls Ingot\ Eye/*.png
Ingot Eye/TENTACLE - Ingot Eye - cover.png

$ ls Void\ Abyss/*.jpg
Void Abyss/cover.jpg
```
As should be expected, only `Void Abyss` displays cover art correctly. If I convert the png file in `Ingot Eye` to jpg it starts to work.

**Comment by Alexey-Yakovenko** [comment]
I will need to test this, probably a regression.. This is supposed to work.

**Comment by Alexey-Yakovenko** [comment]
I can't seem to find a way to download your albums for free (I'm not going to buy them), but I created a test PNG file in gimp, dropped it into a random album's folder, as `cover.png`, and it works. So I'm really sorry, but your bug report is invalid. I used a recent build from git master branch.

**Comment by Alexey-Yakovenko** [comment]
Could be that this is a duplicate of #1131?
Closing either way, cause can't reproduce.

**Comment by orbea** [comment]
You can get the albums for free if you click "Buy Now" and enter 0. If that still does not work I will upload some Lingouf albums somewhere as the creative commons license should allow that.

I have tried creating a PNG file in gimp and dropped it into an album folder titled `cover.png` and it did not work. However if I save it as a JPG file and title it `cover.jpg` it does work...

**Comment by Alexey-Yakovenko** [comment]
> I have tried creating a PNG file in gimp and dropped it into an album folder and titled it cover.png, it did not work. However if I save it as a JPG file and title it cover.jpg it does work...

After the album has been cached as "without album art" -- you need to reset the cache for the newly added cover to be picked up. Deadbeef doesn't attempt loading covers every time a playlist is rendered, for performance reasons. Try deleting ~/.cache/deadbeef, and restarting deadbeef - then see if cover.png works.

If it still doesn't -- means your build is broken. Try using the one which is not broken, e.g. [this one](https://sourceforge.net/projects/deadbeef/files/deadbeef-static_0.7.2-1_x86_64.tar.bz2/download)

**Comment by orbea** [comment]
I just realized I have made a mistake in testing this. The problem is restricted to the icons in the file browser (0.72) plugin I am using...

It can additionally be solved by editing the plugin's configuration settings and adding some possible file names for coverart. It still does not allow wildcards so anything that is not titled exactly right will fail to display, but that an issue for another place...

Thanks for your time!
